### PR TITLE
describe: add support for --append 

### LINKED ISSUE #7508: FR: jj describe --append --message=...
**Is your feature request related to a problem? Please describe.**

From a set of scripts, I want to execute commands and record the results in the change description, step by step.  I would like for these scripts to be composable and simple (for me to write).

**Describe the solution you'd like**

It would be convenient if `jj describe --append --message=...` would exist.  It would append a line (or multiple lines) to the existing change description.

**Describe alternatives you've considered**

Currently, I record the description pieces into a temporary file and execute `jj describe --message=$(cat tmpfile)` afterwards.  This is inconvenient, because I have to pass the temporary file between scripts, and delete it afterwards.

### PR DESCRIPTION
**Pull request by sidracha**
Fixes #7508

Add the `--append` flag to `jj describe` to allow appending text to the end of the current change description, separated by a paragraph break. If the description is currently empty, the command falls back to the standard `jj describe` behavior.

# Checklist

If applicable:

- [x] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (`README.md`, `docs/`, `demos/`)
- [ ] I have updated the config schema (`cli/src/config-schema.json`)
- [x] I have added/updated tests to cover my changes
- [x] I fully understand the code that I am submitting (what it does,
      how it works, how it's organized), including any code drafted by an LLM.
- [x] For any prose generated by an LLM, I have proof-read and copy-edited with
      an eye towards deleting anything that is irrelevant, clarifying anything
      that is confusing, and adding details that are relevant. This includes,
      for example, commit descriptions, PR descriptions, and code comments.

### CODE REVIEW

**Comment by PhilipMetzger** [comment]
Move the fixes line into the commit description to follow the project style described in https://www.jj-vcs.dev/latest/contributing/#commit-guidelines. Especially this sentence: 

> If the PR closes an open issue or feature request for the project, include the Closes #<issue-number> or Fixes #<issue-number> in the commit description.

Also if you're a Googler add yourself to `paid_contributors.md` under the Alphabet/Google section.

File: cli/src/commands/describe.rs [hunk 1]
```diff
@@ -139,8 +145,24 @@ pub(crate) async fn cmd_describe(
...
+                    // Get the old description and strip trailing whitespace
+                    let mut combined = commit.description().trim_end().to_owned();
+
+                    // Add a paragraph break if the old description wasn't empty
+                    if !combined.is_empty() {
+                        combined.push_str("\n\n");
+                    }
+
+                    // Append the new message
+                    combined.push_str(new_message.trim_start());
```
Comment (joshka) [inline cli/src/commands/describe.rs:160]: `trim_start()` removes content rather than just managing the paragraph boundary. I verified this locally:

```text
jj describe          -m '    indented' → "    indented\n"
jj describe --append -m '    indented' → "indented\n"
```

This means appending to an empty description does not actually fall back to ordinary `describe -m` behavior, and it can corrupt indented command output or Markdown code blocks. Could we preserve the new message verbatim and add this as a regression test?

File: cli/tests/test_describe_command.rs [hunk 2]
```diff
@@ -1123,6 +1123,138 @@ fn test_add_trailer_committer() -> TestResult {
     Ok(())
 }
 
+#[test]
+fn test_describe_append() -> TestResult {
```
Comment (joshka) [inline cli/tests/test_describe_command.rs:1127]: Would it be clearer to split these into smaller behavior-focused tests? This test accumulates state across five different input modes, so a failure in an early append changes every later expected description. Independent tests for the empty/existing-description cases, multiple `-m` values, stdin, and editor behavior would make failures easier to localize and give boundary cases such as leading indentation a natural home.

**Review by joshka — CHANGES_REQUESTED** [review]
The append use case looks justified to me. A generic description-template operation would be more expressive, but it would be a verbose and awkward replacement for the common `--append -m "$result"` scripting operation. I’ve asked on [#7508](https://github.com/jj-vcs/jj/issues/7508#issuecomment-5052987417) whether the earlier template suggestion was intended as an alternative that should block this interface or as complementary longer-term direction.

I found one correctness issue: `trim_start()` removes meaningful leading whitespace and means appending to an empty description does not actually fall back to ordinary `describe -m` behavior. I verified that locally with a focused regression. Otherwise, the implementation follows the existing `describe` flow.
